tests/state_propagation_test.py

```diff
@@ -47,203 +47,147 @@ def test_conditional_fake_merge():
 
 
 def test_conditional_full_merge():
```

Comment:
Sure. I can tell you the reasoning behind this change here, preemptively:
This is was not so much a 'fix' as it was to make sure the tests still test for the same behavior or special cases. Essentially the way these tests were written, they fully relied no the _frontend_ (+ simplification) to translate the code into a very specific state machine and interstate edge / symbol combination to test for specific cases and behaviors. This means that not only is the test susceptible to stop testing for what it's trying to test through (any) changes to the frontend, but also through passes other than ConstantPropagation. In making the CFBlock changes I merely noticed this and thus made [this change's](https://github.com/spcl/dace/commit/35e42723df1c1c255ed3420a6717dd4bce520434) with the primary intention of letting these tests test ConstantPropagation and ConstantPropagation only, making it easier to spot and fix issues that are introduced through changes. This is consequently only indirectly related to the LoopRegions change. There would be other ways of making the tests pass for LoopRegions, this is just my proposal to avoid the issue with whatever next changes we may make.